CustomDimensions.php

```diff
@@ -280,10 +281,13 @@ public function addConversionInformation(&$conversion, $visitInformation, Tracke
 
         foreach ($dimensions as $dimension) {
             $index = (int) $dimension['index'];
-            if ($dimension['scope'] === self::SCOPE_VISIT && in_array($index, $conversionIndexes)) {
+            if (in_array($index, $conversionIndexes)) {
                 $field = LogTable::buildCustomDimensionColumnName($dimension);
 
-                if (array_key_exists($field, $visitInformation)) {
+                if ($dimension['scope'] === self::SCOPE_ACTION && array_key_exists($field, $actionDimensions)) {
+                    $conversion[$field] = $actionDimensions[$field];
+                }
+                else if ($dimension['scope'] === self::SCOPE_VISIT  && array_key_exists($field, $visitInformation)) {
```

Comment:
I just tested it and there is a little problem with it. As you may know we have eg 5 custom dimensions for visit table and 5 dimensions for action table. They are all called `custom_dimension_1` ... `custom_dimension_5`.

In my example I used `&dimension1=foo&dimension2=bar` where `dimension1` maps to `custom_dimension_1` in `visit` scope (visit table) and `dimension2` maps to column `custom_dimension_1` in `action` scope (action table). In this case, it would write only one value into `log_conversion` table as both map to `custom_dimension_1`. 

In `log_conversion` table we have by default actually only 5 dimension slots as well when we would now need 10 slots to have enough space for the visit and action dimensions. Otherwise we would mix visit with action dimensions depending if only a visit dimension is set in the request, only an action dimension or both a visit and an action dimension.

This won't be easily fixable as it requires database updates and then likely further logic changes. It may be a bit hard to understand. The thing is that we have by default dimension indexes from 1 to 10 (`dimension1` to `dimension10`). These indexes are mapped to a database column to either `log_visit` table (`custom_dimension1` to `custom_dimension5` column) or `log_link_visit_action` table (`custom_dimension1` to `custom_dimension5`).

Because in `log_conversion` table we also have only 5 columns by default (`custom_dimension1` to `custom_dimension5`) it is not clear to which custom dimension it actually references. It would need more like a `custom_dimension_index_1` to `custom_dimension_index_10` column. Then we would need to change the archiver to generate reports for goal reports differently and some other logic likely in menu etc. This is quite some work.

I think I would now recommend to change this, if we want, for Piwik 4. And I would likely recommend to @code-penguin to apply this logic rather in a new plugin for now to have this effect already. @mattab any thoughts?